dbms/src/Flash/executeQuery.cpp

```diff
@@ -106,6 +108,24 @@ QueryExecutorPtr doExecuteAsBlockIO(IQuerySource & dag, Context & context, bool
     /// Hold element of process list till end of query execution.
     res.process_list_entry = process_list_entry;
 
+    auto auto_spill_trigger_threshold = context.getSettingsRef().auto_memory_revoke_trigger_threshold.get();
+    auto auto_spill_target_threshold = context.getSettingsRef().auto_memory_revoke_target_threshold.get();
+    /// if query level memory tracker has a limit, then setup auto spill trigger
+    if likely (memory_tracker != nullptr)
+    {
+        if (memory_tracker->getLimit() != 0)
+        {
+            auto auto_spill_trigger = [&dag_context, memory_tracker, auto_spill_trigger_threshold, auto_spill_target_threshold]() {
+                if (memory_tracker->get() > memory_tracker->getLimit() * auto_spill_trigger_threshold)
```

Comment:
`memory_tracker->getLimit() * auto_spill_trigger_threshold` can be calculated outside the lambda, no need to recalculate multiple times

Reply:
OK